pmg/models/emails.py

```diff
@@ -151,46 +152,40 @@ def find_or_create(cls, user, q, content_type=None, committee_id=None):
         return search
 
 
-def send_mandrill_email(subject, from_name, from_email, recipient_users, html, utm_campaign, subaccount=None):
-    """ Send an email using Mandrill, relying on Mandrill's templating system.
+def send_sendgrid_email(subject, from_name, from_email, recipient_users, html, utm_campaign):
+    """ Send an email using Sendgrid, relying on Sendgrid's templating system.
 
     :param subject: email subject
     :param from_name: name of the sender
     :param from_email: email of the sender
     :param recipient_users: array of `User` objects of recipients
     :param html: HTML body of the email
-    :param utm_campaign: Google Analytics campaign (optional)
-    :param subaccount: Mandrill subaccount to use (optional)
+    :param utm_campaign: Google Analytics campaign
     """
-    subaccount = subaccount or app.config['MANDRILL_ALERTS_SUBACCOUNT']
-
-    recipients = [{'email': r.email} for r in recipient_users]
-    merge_vars = [{"rcpt": r.email, "vars": [{"name": "NAME", "content": r.name or 'Subscriber'}]} for r in recipient_users]
-
-    # NBNBNBNB: the email template MUST have a special DIV in it to place the content in.
-    # This gets removed when importing the template into Mandrill from Mailchimp
-    #  <div mc:edit="main"></div>
-
-    template_vars = [
-        {"name": "main", "content": html},
-    ]
-
-    msg = {
-        "subject": subject,
-        "from_name": from_name,
-        "from_email": from_email,
-        "to": recipients,
-        "merge_vars": merge_vars,
-        "track_opens": True,
-        "track_clicks": True,
-        "preserve_recipients": False,
-        "google_analytics_campaign": utm_campaign,
-        "google_analytics_domains": ["pmg.org.za"],
-        "subaccount": subaccount,
-    }
+    sg = sendgrid.SendGridClient(app.config['SENDGRID_API_KEY'], raise_errors=True)
+
+    recipients = [r.email for r in recipient_users]
+
+    message = sendgrid.Mail()
+
+    message.smtpapi.add_to(recipients)
+    message.set_subject(subject)
+    message.add_filter('templates', 'enable', '1')
+    message.add_filter('templates', 'template_id', app.config['SENDGRID_TRANSACTIONAL_TEMPLATE_ID'])
+
+    message.smtpapi.set_substitutions({'*|NAME|*': [r.name or 'Subscriber' for r in recipient_users]})
+
+    message.add_filter('ganalytics', 'enable', '1')
+    message.add_filter('ganalytics', 'utm_medium', 'email')
+    message.add_filter('ganalytics', 'utm_source', 'transactional')
+    message.add_filter('ganalytics', 'utm_campaign', utm_campaign)
+    message.add_filter('clicktrack', 'enable', '1')
+
+    message.set_html(html)
+    message.set_from(from_email)
+    status, msg = sg.send(message)
 
     log.info("Email will be sent to %d recipients." % len(recipients))
-    log.info("Sending email via mandrill: %s" % msg)
+    log.info("Sending email via sendgrid: %s" % msg)
```

Comment:
Sorry, I misread this. I'd like us to log that we're about to send the email just before we send it ("Email will be sent to..."). That provides context if the sg.send call raises an exception.

Then after it's sent we can log the result of the send.

Pls also remove the `print`.